src/runtime/internal/dom.ts

```diff
@@ -112,6 +112,36 @@ function init_hydrate(target: NodeEx) {
 	}
 }
 
+export function append_styles(
+	target: Node,
+	style_sheet_id: string,
+	styles: string
+) {
+	const append_styles_to = get_root_for_styles(target);
+
+	if (!append_styles_to?.querySelector(`#${style_sheet_id}`)) {
+		const style = element('style');
+		style.id = style_sheet_id;
+		style.textContent = styles;
+		append(append_styles_to, style);
+	}
+}
+
+export function get_root_for_node(node: Node) {
+	if (!node) return document;
+
+	return (node.getRootNode ? node.getRootNode() : node.ownerDocument); // check for getRootNode because IE is still supported
```

Comment:
@Conduitry does svelte still support IE? Or can we remove this part?

Reply:
I think this is a good check to be doing. We're not supporting IE out of the box, but we'd prefer to not do anything that can't be transpiled or polyfilled for IE in a reasonably straightforward way.